lib/fui/finder.rb

```diff
@@ -51,11 +51,24 @@ def process_code(references, path)
         filename = File.basename(path)
         headers.each do |header|
           filename_without_extension = File.basename(path, File.extname(path))
-          references[header] << path if filename_without_extension != header.filename_without_extension && File.read(file).include?("#import \"#{header.filename}\"")
+          file_contents = File.read(file)
+          global_import_exists = global_imported(file_contents, header)
+          local_import_exists = local_imported(file_contents, header)
+          references[header] << path if filename_without_extension != header.filename_without_extension && (local_import_exists || global_import_exists)
         end
       end
     end
 
+    def local_imported(file_contents, header)
+      file_contents.include?("#import \"#{header.filename}\"")
+    end
+
+    def global_imported(file_contents, header)
+      escaped_header = Regexp.quote(header.filename)
+      regex = '(#import\s{1}<.+\/' + escaped_header + '>)'
+      options['global'] && file_contents.match(regex)
```

Comment:
We run the regex even if `!options['global']` which seems wasteful, add an `if` in the finder method above to skip `global_imported`.

Reply:
Updated.